flopy/mf6/data/mfdataplist.py

```diff
@@ -0,0 +1,2527 @@
+                Verify data prior to storing
+
+        """
+        self._cache_model_grid = True
+        for sp in self._data_storage.keys():
+            self._current_key = sp
+            storage = self._get_storage()
+            if storage.internal_size == 0:
+                storage.internal_data = self.get_dataframe(sp)
+            if storage.internal_size > 0 and (
+                self._get_storage().data_storage_type
+                != DataStorageType.external_file
+                or replace_existing_external
+            ):
+                fname, ext = os.path.splitext(external_file_path)
+                if datautil.DatumUtil.is_int(sp):
+                    full_name = f"{fname}_{int(sp) + 1}{ext}"
+                else:
+                    full_name = f"{fname}_{sp}{ext}"
+
+                super().store_as_external_file(
+                    full_name,
+                    binary,
+                    replace_existing_external,
+                    check_data,
+                )
+        self._cache_model_grid = False
+
+    def store_internal(
+        self,
+        check_data=True,
+    ):
+        """Store all data internally.
+
+        Parameters
+        ----------
+            check_data : bool
+                Verify data prior to storing
+
+        """
+        self._cache_model_grid = True
+        for sp in self._data_storage.keys():
+            self._current_key = sp
+            if (
+                self._get_storage().data_storage_type
+                == DataStorageType.external_file
+            ):
+                super().store_internal(
+                    check_data,
+                )
+        self._cache_model_grid = False
+
+    def has_data(self, key=None):
+        """Returns whether this MFList has any data associated with it in key
+        "key"."""
+        if key is None:
+            for sto_key in self._data_storage.keys():
+                self.get_data_prep(sto_key)
+                if super().has_data():
+                    return True
+            return False
+        else:
+            self.get_data_prep(key)
+            return super().has_data()
+
+    def has_modified_ext_data(self, key=None):
+        if key is None:
+            for sto_key in self._data_storage.keys():
+                self.get_data_prep(sto_key)
+                if super().has_modified_ext_data():
+                    return True
+            return False
+        else:
+            self.get_data_prep(key)
+            return super().has_modified_ext_data()
+
+    def binary_ext_data(self, key=None):
+        if key is None:
+            for sto_key in self._data_storage.keys():
+                self.get_data_prep(sto_key)
+                if super().binary_ext_data():
+                    return True
+            return False
+        else:
+            self.get_data_prep(key)
+            return super().binary_ext_data()
+
+    def get_record(self, key=None, data_frame=False):
```

Comment:
This needs to be consistent with the existing interface (MFList and MFArray).  We can discuss renaming the interface across all data classes, but we need to be careful in doing so since it will break user's existing code.  One goal of this PR is to implement Pandas Lists without breaking our existing interface.  Perhaps we can discuss this for a future PR?